File: challenge1/Python/countSequences.py
n, S = map(int, input().split())

l = [0] * (n)
r = [0] * (n)

for i in range(0, n): 
    l[i], r[i] = map(int, input().split())
    
def solve(idx, total): 
    if idx == n: 
        if total == S: 
            return 1
        else:
            return 0
    res = 0
    for val in range(l[idx], r[idx] + 1): 
        res += solve(idx + 1, total + val)
    return res

print(solve(0, 0))

File: challenge1/C++/countSequences.cpp
#include <iostream>
#include <vector>
#include <functional>

int main() {

  int n, S;
  std::cin >> n >> S;

  std::vector<int> l(n), r(n);
  for (int i = 0; i < n; i++) 
    std::cin >> l[i] >> r[i];

  std::function<int(int, int)> solve = [&](int index, int total) {
    if (index == n) 
      return (total == S ? 1 : 0);
    int res = 0;
    for (int value = l[index]; value <= r[index]; value++) 
      res += solve(index + 1, total + value);
    return res;
  };

  std::cout << solve(0, 0);

  return 0;
}